src/components/visualizations/utils/drilldownEventing.ts

```diff
@@ -81,58 +88,52 @@ function fireEvent(onFiredDrillEvent: OnFiredDrillEvent, data: IDrillEvent, targ
     }
 }
 
-function normalizeIntersectionElements(intersection: ILegacyDrillIntersection[]): IDrillEventIntersectionElement[] {
-    return intersection.map(({ id, title, value, name, uri, identifier }) => {
-       const intersection: IDrillEventIntersectionElement = {
-           id,
-           title: title || value as string || name
-       };
-       if (uri || identifier) {
-           intersection.header = { uri, identifier };
-       }
-       return intersection;
-    });
-}
-
 function composeDrillContextGroup(
     { points }: IHighchartsChartDrilldownEvent,
-    chartType: VisType
+    chartType: ChartType
 ): IDrillEventContextGroup {
+    const contextPoints: IDrillPoint[] = points.map((point: IHighchartsPointObject) => {
+        return {
+            x: point.x,
+            y: point.y,
+            intersection: point.drillIntersection
+        };
+    });
     return {
         type: chartType,
-        element: 'label',
-        points: points.map((point: IHighchartsPointObject) => {
-            return {
-                x: point.x,
-                y: point.y,
-                intersection: normalizeIntersectionElements(point.drillIntersection)
-            };
-        })
+        element: VisElementTypes.LABEL,
+        points: contextPoints
     };
 }
 
 function composeDrillContextPoint(
     { point }: IHighchartsChartDrilldownEvent,
-    chartType: VisType
-): DrillEventContext {
-    const zProp = isNaN(point.z) ? {} : { z: point.z };
-    const valueProp = (isTreemap(chartType) || isHeatmap(chartType)) ? { value: point.value.toString() } : {};
-    const xyProp = isTreemap(chartType) ? {} : {
-        x: point.x,
-        y: point.y
-    };
-    return {
+    chartType: ChartType
+): IDrillEventContextPoint {
+    const context: IDrillEventContextPoint = {
         type: chartType,
         element: getClickableElementNameByChartType(chartType),
-        ...xyProp,
-        ...zProp,
-        ...valueProp,
-        intersection: normalizeIntersectionElements(point.drillIntersection)
+        intersection: point.drillIntersection
```

Comment:
`normalizeIntersectionElements` is not necessary ?

Reply:
1. Removed `ILegacyDrillIntersection` and `normalizeIntersectionElements`. Charts and Table now use correct `IDrillEventIntersectionElement` interface internally.